internal/api/e2e/client/test_utils.go

```diff
@@ -22,11 +22,21 @@ import (
 	"github.com/perses/perses/internal/api/dependency"
 	e2eframework "github.com/perses/perses/internal/api/e2e/framework"
 	"github.com/perses/perses/pkg/client/api/v1"
-	"github.com/perses/perses/pkg/client/perseshttp"
+	"github.com/perses/perses/pkg/client/config"
 	"github.com/perses/perses/pkg/model/api"
 	"github.com/perses/perses/pkg/model/api/v1/common"
 )
 
+func withAuthClient(t *testing.T, testFunc func(v1.ClientInterface, dependency.PersistenceManager) []api.Entity) {
+	server, _, persistenceManager := e2eframework.CreateServer(t, e2eframework.DefaultAuthConfig())
+	defer server.Close()
+	persesClient := createAuthClient(t, server)
+	usrEntity := e2eframework.NewUser("foo", "$2a$10$iCemKjvjN.ieqJOPlEL5keGRLAGKRGBNjph2N8uY.4XFPZYwcBT8K") // the encrypted value for password
```

Comment:
too precise to deserve a test_utils function in my opinion. the withClient as well. 

You can directly use the withServer config. It provides now the URL in the ``testFunc``
e.g https://github.com/perses/perses/blob/a403f6433958d96f85d5e80c8492c4bc03fb484f/internal/api/e2e/api/auth_test.go#L437-L450

Reply:
I would even remove the ``withClient`` function

Reply:
withClient simplify the creation of the client to avoid to do it for every tests.

Reply:
I have simplifies the oauth tests ;)